runtime/src/sorted_storages.rs

```diff
@@ -45,21 +45,25 @@ impl<'a> SortedStorages<'a> {
                 storage.slot() // this must be unique. Will be enforced in new_with_slots
             })
             .collect::<Vec<_>>();
-        Self::new_with_slots(source, &slots)
+        Self::new_with_slots(source, &slots, None)
     }
 
     // source[i] is in slot slots[i]
     // assumptions:
     // 1. slots vector contains unique slot #s.
     // 2. slots and source are the same len
-    pub fn new_with_slots(source: &'a [SnapshotStorage], slots: &[Slot]) -> Self {
+    pub fn new_with_slots(
+        source: &'a [SnapshotStorage],
+        slots: &[Slot],
+        slot: Option<Slot>,
```

Comment:
Could we add some documentation for what this argument is and what `Some()` vs `None` means for this particular argument?

If this `slot` the maximum slot in the `slots`slice, does it seem appropriate to rename `slot` to `max_slot`?